03.2_school_prepare_OD_file.R

```diff
@@ -8,14 +8,27 @@ geography <- "lsoa"
 set.seed(12345)
 
 #########################
-# LOAD SCHOOL FLOW DATASET
+# LOAD SCHOOL FLOW DATASETS
 #########################
-sf11 <- read_tsv(file.path(path_inputs, "01_raw/02_travel_data/school/NPD_originals_PRIVATE/Spring_Census_2011.txt"))
-sf11 <- sf11[,c(1, 3:4, 6:10)]
-names(sf11) <- sub(paste0("_SPR11"), "", names(sf11))
-names(sf11) <- tolower(names(sf11))
-sf11 <- dplyr::rename(sf11, lsoa01cd = `llsoa`, bicycle = `cycle`, foot = `walk`)
+# Open and rename the 2010 and 2011 datasets
+for (i in (10:11) ) {
+  sf <- read_tsv(file.path(path_inputs, "01_raw/02_travel_data/school/lsoa/NPD_originals_PRIVATE", paste0("Spring_Census_20",i,".txt")))
+  sf <- sf[,c(1, 3:4, 6:10)]
+  names(sf) <- sub(paste0("_SPR",i), "", names(sf))
+  names(sf) <- tolower(names(sf))
+  sf <- dplyr::rename(sf, lsoa01cd = `llsoa`, bicycle = `cycle`, foot = `walk`)
+  assign(paste0("sf",i), data.frame(sf))
+}
 
+# Drop from 2011 the flows with too much missing data (0.3%) in 2011 but enough in 2010 , and rebind
+# this list was derived semi-manually downstream in '05.2 EXTRA' code commented out: to redo it, have to comment out this subsection and re-run that code
+sf11_missing <- read_csv(file.path(path_inputs, "01_raw/02_travel_data/school/lsoa/x-manual_extras", "0_flowdata_missing2011_present2010.csv"))
+sf11 <- left_join(sf11, sf11_missing, by="urn")
+sf10 <- left_join(sf10, sf11_missing, by="urn")
+sf11 <- sf11[is.na(sf11$replace2011),]
+sf10 <- sf10[!is.na(sf10$replace2011),]
+sf11 <- rbind(sf11, sf10)
```

Comment:
Sure it works but suggest using **tidyverse** equivalents of base functions for consistency. Likewise with `read_tsv()`

Reply:
sorry, am not sure which function you are suggesting replacing – happy to switch to a tidy verse equivalent